src/detect-tls.c

```diff
@@ -676,35 +551,16 @@ static int DetectTlsFingerprintMatch (DetectEngineThreadCtx *det_ctx,
  */
 static int DetectTlsFingerprintSetup (DetectEngineCtx *de_ctx, Signature *s, const char *str)
 {
-    DetectTlsData *tls = NULL;
-    SigMatch *sm = NULL;
 
-    if (DetectSignatureSetAppProto(s, ALPROTO_TLS) != 0)
+    if (DetectContentSetup(de_ctx, s, str) < 0) {
```

Comment:
I don't know if it will bring any advantages, but since mpm is aware of depth we might set and implied depth to anchor the patterns to the start of the buffer. At least with our Aho-Corasick implementation there may be an advantage to it

Reply:
HS also aggressively optimizes, so perhaps it also has extra tricks for a case where all patterns in the set anchor to the start of the buffer

Reply:
@jlucovsky can you do optimization ticket for this so we don't forget?